### ISSUE TITLE
ACF Pro 6.0.0-RC1 - can't edit InnerBlocks until you enter and exit the block's Edit mode (regression from beta 1)

### ISSUE THREAD

**Issue by ericestes** [AdvancedCustomFields/acf#700]
With a simple block containing an `<InnerBlocks />`, the inner blocks are locked and cannot be edited until you enter and exit the block's edit mode. This happens with or without fields being assigned to the block.

block.json:
```
{
  "$schema": "https://schemas.wp.org/trunk/block.json",
  "apiVersion": 2,
  "name": "testing/test",
  "title": "Test",
  "acf": {
    "mode": "preview",
    "renderTemplate": "./template.php"
  },
  "supports": {
    "jsx": true
  }
}
```

template.php:
```
<div style="min-height: 50px; outline: 1px solid red">
    <InnerBlocks />
</div>
```

**Comment by CreativeDive** [comment]
@ericestes thanks for that. This is already a known bug in RC1 which will be fixed by @lgladdy in the next RC version.

**Comment by ericestes** [comment]
Thanks @CreativeDive and @lgladdy!

**Comment by lgladdy** [comment]
@ericestes We should have an RC2 out later this week which fixes this - so please yell if you still have issues with that build!

**Comment by ericestes** [comment]
@lgladdy @CreativeDive another issue I've been seeing on beta 1 and RC1 is InnerBlocks will save correctly but when you re-open the editor they'll lose content. One experience I've had is the first inner block's content will be retained but any following blocks are lost and in their place you see the original inner block template.

Is this something you're already aware of and/or related to incoming bugfix you mentioned? If not, I'm happy to open a new ticket and try to find a reproducible example.

**Comment by lgladdy** [comment]
@ericestes Yeh I think it's probably related. There's issues with preloaded blocks replacing InnerBlocks with server side rendered HTML. If you drop an email to support@advancedcustomfields.com FAO Liam, I'll send you the build we think fixes this ahead of RC2 later this week, and you can help us test if you like?

**Comment by ericestes** [comment]
> @ericestes Yeh I think it's probably related. There's issues with preloaded blocks replacing InnerBlocks with server side rendered HTML. If you drop an email to [support@advancedcustomfields.com](mailto:support@advancedcustomfields.com) FAO Liam, I'll send you the build we think fixes this ahead of RC2 later this week, and you can help us test if you like?

Will do! Thanks agin

**Comment by dnnsjsk** [comment]
Also experiencing this! RC2 didn't fix it for me

**Comment by lgladdy** [comment]
@dnnsjsk Could you provide some more info and ideally some code examples? Eric's issue turns out to be a WordPress issue with block locking.

**Comment by dnnsjsk** [comment]
@lgladdy 
Sure thing:
[acf-test.zip](https://github.com/AdvancedCustomFields/acf/files/9491319/acf-test.zip)

1. Install plugin.
2. Go to editor, insert "Spalten"
3. Click into it and insert another block (it only allowed one block)
4. Save and reload

Bit of a different setup than the thread states since there is an InnerBlock enabled ACF block inside another InnerBlock enabled ACF block.

**Comment by dnnsjsk** [comment]
Also, another bug:
![PixelSnap 2022-09-05 at 18 55 52](https://user-images.githubusercontent.com/73965001/188492217-462591f3-6dc9-4cc8-b910-f72e588fdd28.png)

There is no content in the second "Spalten" block, however it copies the content of the first "Spalten" block? It does away after pressing Edit Mode -> Preview again.

Also the content in the first "Spalten" block is wrong, it should look like this:
![PixelSnap 2022-09-05 at 18 57 43](https://user-images.githubusercontent.com/73965001/188492392-fd54b1a4-fc88-438d-992a-1080c675b162.png)

I fixed it by doing the Edit->Preview mode for the respective blocks in the second screenshot.

**Comment by dnnsjsk** [comment]
Ahh I see. Yes, that fixed it for me, I'll leave it off for now. Thanks!

**Comment by lgladdy** [comment]
@dnnsjsk Cheers! You raise an interesting point though, there might be other people with this who'll be affected so we'll note this in the release notes at a minimum, but ideally try and incorporate something about the inner blocks content as part of the hashing.

We didn't really expect people to use blocks with any ACF fields on the block!

**Comment by CreativeDive** [comment]
@lgladdy after updating to ACR 6.0 RC2 I have a lot of issues with inner blocks. I'm just trying to figure out what's causing this.

**Comment by lgladdy** [comment]
@CreativeDive That's interesting! There shouldn't be any differences between RC2 and the build you got last week in theory for blocks!

**Comment by CreativeDive** [comment]
@lgladdy I hadn't worked with RC1 because I couldn't edit inner blocks. So I can't make a comparison between RC1 and RC2.

**Comment by lgladdy** [comment]
I think I've been able to reproduce this bug with not being able to edit preloaded InnerBlocks. Leave it with me and I'll get back to you with an update soon.

**Comment by CreativeDive** [comment]
@lgladdy Here one example:

It's my row block, and each row item is using an inner block. It worked correctly before updating to RC2.

<img width="1264" alt="Bildschirmfoto 2022-09-05 um 20 02 14" src="https://user-images.githubusercontent.com/31563167/188498538-3fbcff88-0b2c-4c0c-baab-af4f68b60593.png">

But the inner block are not selectable now with RC2.

Here the video with the latest beta version (everything was fine):

https://user-images.githubusercontent.com/31563167/188499655-7117cb5d-8b1d-4821-9fdf-96c2228d40d5.mp4

Here the video with RC2:

https://user-images.githubusercontent.com/31563167/188499376-fa3b5997-3169-4f7c-83da-b01ad23e5fb3.mp4

Change to RC2, refreshing the page and you can see, I can't select an inner block and "ROW 3" is now "ROW 2". And adding a new row block should be starting empty like in video number 1. This is really confusing bug.

**Comment by lgladdy** [comment]
@CreativeDive Can you ditch the `doing_filter` code we added to try and support blocks that render other blocks? I think we might have to ditch that from 6.0 and come back to it in 6.0.1.

So make line 573 of pro/blocks.php:
```php
	if ( ! $is_preview ) {
```

**Comment by CreativeDive** [comment]
> So make line 573 of pro/blocks.php:

@lgladdy Yes it seems that is fixing this issue. But unfortunately the rendering of block patterns in the editor no longer works. Unfortunately, that doesn't seem to be as easy as expected. But it would be nice if this could also work.

**Comment by lgladdy** [comment]
@CreativeDive Yeh, we'll come back to it soon after 6.0's release. Just to confirm, the block pattern rendering was also an issue before we added the code in RC1 to try and fix it right? So, with that code change on line 573, we're back to Beta 1's behaviour?

**Comment by CreativeDive** [comment]
@lgladdy yes I think this is the behaviour of the latest beta. I will continue my testing in the next days with RC2.

But all you will do with:

`apply_filters( 'the_content', $block_content );` OR filter `render_block` OR filter `do_blocks` currently does not work when trying to convert `block patterns to fully rendered blocks` or trying to modify post_content via `parse_blocks()` by using the `render_block` filter. This is just not working in the block editor context.

The same behaviour like in the latest beta.

**Comment by lgladdy** [comment]
@CreativeDive Yeh, the issue is the way we preload blocks is due to a quirk of WordPress. It basically renders the page as if it's the front-end, so those filters would be true for the preload too.

We need to be able to separate internal WordPress fake-REST calls (which is how preloading works) from other calls rather than relying on filter checks. We've already written code to set flags for REST calls to fix issues with script loading, but we need more time to test switching to that system here, and I'd prefer to run more test builds with everyone here on this.

**Comment by CreativeDive** [comment]
@lgladdy Yes I'm here to test to help.

**Comment by CreativeDive** [comment]
@lgladdy Another idea, but I don't know if this could work is to add an ACF filter like `doing_filter('acf_render_inner_blocks')`. We could manually use such a filter to output the frontend rendering in the block editor.

**Comment by mwhiteley16** [comment]
Can also confirm issue of not being able to edit inner block after initial save. RC2 did not fix for me. Let me know if there is anything I can do to help.

**Comment by lgladdy** [comment]
@mwhiteley16 Does this happen if you replace line 573 as per this comment: https://github.com/AdvancedCustomFields/acf/issues/700#issuecomment-1237374553 too?

**Comment by CreativeDive** [comment]
@lgladdy Or would something like that be conceivable?

```
remove_filter( 'the_content', 'acf_render_inner_blocks_for_editor' );

Between this we can output our converted block in the frontend way.

add_filter( 'the_content', 'acf_render_inner_blocks_for_editor' );
```

I'm just wondering if there might be another way.

**Comment by lgladdy** [comment]
Hey Everyone,

This issue should now be resolved in ACF 6.0.0-RC3. Please let me know if you're still having issues!
